pkg/util/cloudfoundry/bbscache_test.go

```diff
@@ -0,0 +1,172 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+// +build clusterchecks
+
+package cloudfoundry
+
+import (
+	"os"
+	"testing"
+	"time"
+
+	"code.cloudfoundry.org/bbs/events"
+	"code.cloudfoundry.org/bbs/models"
+	"code.cloudfoundry.org/lager"
+	"github.com/stretchr/testify/assert"
+)
+
+type testBBSClient struct {
+}
+
+func (t testBBSClient) ActualLRPs(lager.Logger, models.ActualLRPFilter) ([]*models.ActualLRP, error) {
+	return []*models.ActualLRP{&BBSModelA1, &BBSModelA2}, nil
+}
+
+func (t testBBSClient) DesiredLRPs(lager.Logger, models.DesiredLRPFilter) ([]*models.DesiredLRP, error) {
+	return []*models.DesiredLRP{&BBSModelD1}, nil
+}
+
+var c *BBSCache
+
+func TestMain(m *testing.M) {
+	c, _ = ConfigureGlobalBBSCache("url", "", "", "", time.Second, true)
+	c.bbsAPIClient = testBBSClient{}
+	for range []int{0, 1} {
+		if c.GetPollSuccesses() == 0 {
+			time.Sleep(time.Second)
+		}
+	}
+	code := m.Run()
+	os.Exit(code)
+}
+
+func TestBBSCachePolling(t *testing.T) {
+	assert.NotZero(t, c.GetPollAttempts())
+	assert.NotZero(t, c.GetPollSuccesses())
+}
+
+func TestBBSCache_GetDesiredLRPs(t *testing.T) {
+	assert.EqualValues(t, []DesiredLRP{ExpectedD1}, c.GetDesiredLRPs())
+}
+
+func TestBBSCache_GetActualLRPs(t *testing.T) {
+	assert.EqualValues(t, []ActualLRP{ExpectedA1, ExpectedA2}, c.GetActualLRPsFor("012345678901234567890123456789012345"))
+}
+
+func TestBBSCache_GetAllLRPs(t *testing.T) {
+	a, d := c.GetAllLRPs()
+	assert.EqualValues(t, ([]DesiredLRP{ExpectedD1}), d)
+	assert.EqualValues(t, map[string][]ActualLRP{"012345678901234567890123456789012345": {ExpectedA1, ExpectedA2}}, a)
+}
+
+// These methods follow to ensure we implement the bbs.Client API, but are in fact unused
+func (t testBBSClient) DesireTask(logger lager.Logger, guid, domain string, def *models.TaskDefinition) error {
```

Comment:
`These methods follow to ensure` => `These methods ensure`
Is this work to do later?

Reply:
It's not work to do later; it's just that we have these methods to have the testing BBS client to actually implement the interface, but these methods are never really used by us, so we just leave them blank.